components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java

```diff
@@ -258,8 +260,19 @@ public static boolean isValidCaptcha(String reCaptchaResponse) throws CaptchaExc
         try {
             try (InputStream in = entity.getContent()) {
                 JsonObject verificationResponse = new JsonParser().parse(IOUtils.toString(in)).getAsJsonObject();
-                if (verificationResponse == null || verificationResponse.get("success") == null ||
-                        !verificationResponse.get("success").getAsBoolean()) {
+                String score = verificationResponse.get("score").getAsString();
+                if (log.isDebugEnabled()) {
+                    if (StringUtils.isNotBlank(score)) {
+                        log.debug("reCAPTCH v3 success:" + verificationResponse.get("success") + ", action:" +
```

Comment:
have the else block other way

if (StringUtils.isNotBlank(score)) {
//Recaptch V3 response constains score
if(llog.isDebugEnabled()){  log.debug() }
} else {
if(llog.isDebugEnabled()){ log.debug() }
}